storage_controller/src/http.rs

```diff
@@ -89,6 +91,46 @@ fn get_state(request: &Request<Body>) -> &HttpState {
         .as_ref()
 }
 
+/// Rate limits tenant requests.
+///
+/// TODO: this should be a request middleware, but requires us to extract the tenant ID from
+/// different URLs in a systematic way.
+///
+/// TODO: consider returning a 429 response if these start piling up.
+///
+/// TODO: consider adding a config option for the rate.
+async fn maybe_rate_limit(tenant_id: TenantId) {
+    // Check if the tenant should be rate-limited.
+    const TENANT_RATE_LIMIT: governor::Quota =
+        governor::Quota::per_second(NonZero::new(10).unwrap());
```

Comment:
Is 10 per second reasonable for all these endpoints?

Reply:
We only hit this 3 times across all of our integration tests. Will run it in staging for a bit and see.